### ISSUE TITLE
V6 fix: bump fast-xml-parser to v4.4.1 for V6

### ISSUE THREAD

**Issue by ashika112** [aws-amplify/amplify-js#13670]
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-js/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-js/discussions).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### JavaScript Framework
ALL

### Amplify Version
v6

### Describe the bug

Amplify Version : `6.4.2`

`yarn audit`

![image](https://github.com/user-attachments/assets/2cc27dd8-984b-4c77-9938-04fac91045aa)

### Solution
Update `aws-sdk-js-v3` to latest (3.621.0)

### Reproduction steps

`yarn audit`

### Additional information and screenshots
Github advisories - https://github.com/advisories/GHSA-mpg4-rc92-vx8v
AWS SDK JS V3 Ref - https://github.com/aws/aws-sdk-js-v3/issues/6331

**Comment by ashika112** [comment]
We are currently working on a fix in the PR - https://github.com/aws-amplify/amplify-js/pull/13665

This is coming from a downstream dependency from `aws-sdk-js-v3` and needs to updated to latest version for the sdk. The PR addresses this and is going through testing.

**Comment by ryanknight09** [comment]
Seeing the same issue for Amplify in my current project. Thanks for working on this. 👏

**Comment by bhaveshj459** [comment]
Hey ca anyone help me with this

**Comment by maurer2** [comment]
> Hey can anyone help me with this
> 
> fast-xml-parser <4.4.1 Severity: high fast-xml-parser vulnerable to ReDOS at currency parsing - [GHSA-mpg4-rc92-vx8v](https://github.com/advisories/GHSA-mpg4-rc92-vx8v) fix available via `npm audit fix --force` Will install aws-amplify@5.3.20, which is a breaking change node_modules/@aws-sdk/client-sts/node_modules/fast-xml-parser @aws-sdk/client-sts <=3.525.0 Depends on vulnerable versions of fast-xml-parser node_modules/@aws-sdk/client-sts @aws-sdk/client-firehose 3.12.0 - 3.525.0 Depends on vulnerable versions of @aws-sdk/client-sts node_modules/@aws-sdk/client-firehose @aws-sdk/client-kinesis 3.12.0 - 3.525.0 Depends on vulnerable versions of @aws-sdk/client-sts node_modules/@aws-sdk/client-kinesis @aws-sdk/client-personalize-events 3.12.0 - 3.525.0 Depends on vulnerable versions of @aws-sdk/client-sts node_modules/@aws-sdk/client-personalize-events @aws-amplify/analytics >=7.0.0 Depends on vulnerable versions of @aws-sdk/client-firehose Depends on vulnerable versions of @aws-sdk/client-kinesis Depends on vulnerable versions of @aws-sdk/client-personalize-events node_modules/@aws-amplify/analytics aws-amplify >=6.0.0 Depends on vulnerable versions of @aws-amplify/analytics node_modules/aws-amplify @aws-amplify/ui <=0.0.0-theming-v2-f8422a3-20240627193426 || >=6.0.0 Depends on vulnerable versions of aws-amplify node_modules/@aws-amplify/ui @aws-amplify/ui-angular <=0.0.0-theming-v2-f8422a3-20240627193426 || >=5.0.0 Depends on vulnerable versions of @aws-amplify/ui Depends on vulnerable versions of aws-amplify node_modules/@aws-amplify/ui-angular

Hello, as a temporary solution you can add the override for `fast-xml-parser` in the package.json until a new version is available:

When using npm
```
"overrides": {
    "fast-xml-parser": "4.4.1"
},
```

When using yarn
```
"resolutions": {
    "fast-xml-parser": "4.4.1"
},
```
When using pnpm
```
"pnpm": {
    "overrides": {
        "fast-xml-parser": "4.4.1"
    }
}

```

I believe bun has support for the "overrides" and "resolutions" fields, so the either of the first two solution should work.

Cheers

**Comment by cranberyxl** [comment]
@ashika112 `@aws-amplify/analytics` is still at version `7.0.40` https://github.com/aws-amplify/amplify-js/blame/main/packages/analytics/package.json#L3 which doesn't pickup the changed `aws-sdk` versions

**Comment by ashika112** [comment]
@cranberyxl Sorry merging my PR closed this issue. we still need to do a release. Are you utilizing `unstable` of aws-amplify, only `unstable` has the changes as of now and needs to be released as new version!

**Comment by cranberyxl** [comment]
My bad. I thought since it was closed it was released.

**Comment by ashika112** [comment]
Fix is now released in `aws-amplify @ 6.4.4`
ref: https://www.npmjs.com/package/aws-amplify/v/6.4.4
